src/reanimated2/globals.d.ts

```diff
@@ -1,143 +1,96 @@
+/* eslint-disable no-var, @typescript-eslint/no-explicit-any */
 import type {
   AnimatedStyle,
   StyleProps,
   MeasuredDimensions,
   MapperRegistry,
   ShareableRef,
   ShareableSyncDataHolderRef,
+  ShadowNodeWrapper,
 } from './commonTypes';
 import type { FrameCallbackRegistryUI } from './frameCallback/FrameCallbackRegistryUI';
-import type { ShadowNodeWrapper } from './hook/commonTypes';
-import { LayoutAnimationStartFunction } from './layoutReanimation';
 import type { NativeReanimated } from './NativeReanimated/NativeReanimated';
+import type {
+  LayoutAnimationFunction,
+  LayoutAnimationType,
+  LayoutAnimationsValues,
+} from './layoutReanimation/animationBuilder';
 
 declare global {
-  const _WORKLET: boolean;
-  const _IS_FABRIC: boolean;
-  const _REANIMATED_VERSION_CPP: string;
-  const _REANIMATED_VERSION_BABEL_PLUGIN: string;
-  const __reanimatedModuleProxy: NativeReanimated;
-  const evalWithSourceMap: (
+  var _WORKLET: boolean;
+  var _IS_FABRIC: boolean;
+  var _REANIMATED_VERSION_CPP: string;
+  var _REANIMATED_VERSION_BABEL_PLUGIN: string;
+  var __reanimatedModuleProxy: NativeReanimated;
```

Comment:
We can also update these types, since all of these are `undefined` on web.
```suggestion
  var _WORKLET: boolean | undefined;
  var _IS_FABRIC: boolean | undefined;
  var _REANIMATED_VERSION_CPP: string | undefined;
  var _REANIMATED_VERSION_BABEL_PLUGIN: string | undefined;
  var __reanimatedModuleProxy: NativeReanimated | undefined;
```
Also, I'm wondering if TypeScript supports platform extensions, e.g. `.web.ts`, so we can use different types on native and web (probably not).

Reply:
It seems to be [doable](https://stackoverflow.com/questions/44001050/platform-specific-import-component-in-react-native-with-typescript) but not the simplest.

Reply:
The example you've linked to seems to be a bit different - there are two implementations (Android, iOS) which have the same type, and Metro bundler can choose them based on the suffix (e.g. `.android.tsx`). In our case, the web implementation has different types than the native one and it looks like TypeScript cannot into suffixes.

Reply:
For now, adding `| undefined` is fine I think.